### ISSUE TITLE
Storage gets cleared every other day (iOS)

### ISSUE THREAD

**Issue by danielehrhardt** [supabase/supabase-js#1317]
# Bug report
https://forum.ionicframework.com/t/storage-gets-cleared-every-other-day-ios14/203254
The Supabase token are in the local storage the problem is that on Ionic Capacitor Apps the iOS Localstorage will be cleared every other day. This is why it need to be saved somewhere else. StorageDB for example.

**Comment by kewur** [comment]
I have the same issue with angular 18

**Comment by danielehrhardt** [comment]
> I have the same issue with angular 18. I am using a custom storage, but storage works fine for one day. the next day, on the first load the stored key isnt accessed (says last accessed the day before) and on page refresh the stored key gets deleted altogether.
```

 const customStorage = {
      getItem: (key: string) => this.storage.get(key),
      setItem: (key: string, value: string) => this.storage.set(key, value),
      removeItem: (key: string) => this.storage.remove(key),
    };

    this.supabase = createClient(environment.supabaseUrl, environment.supabaseKey, {
      auth: {
        storage: customStorage,
      },
    });
```

**Comment by kewur** [comment]
doesnt work with angular. I'll create a new ticket then, thanks for the reply anyway

**Comment by danielehrhardt** [comment]
> doesnt work with angular. I'll create a new ticket then, thanks for the reply anyway

The Code is from a angular app

**Comment by kewur** [comment]
cant use this.storage since that doesnt work with ssr.

**Comment by danielehrhardt** [comment]
> cant use this.storage since that doesnt work with ssr.

you need to add your own indexdb logic

**Comment by galaxyblur** [comment]
Was this solved for anyone? I haven't been able to get a custom storage solution working. On top of that, is that the solution that the Supabase team recommends?

I could understand if Supabase did not support this type of integration (web->native app wrapper via Ionic and Capacitor), but there are literally docs for performing this integration on Supabase's own site. As it stands, those integrations are incomplete because they do not offer a persistent auth solution.

**Comment by orleib-lab** [comment]
Related app-side pattern, not a substitute for durable storage on this issue.

iOS WebView `localStorage` eviction (this thread) and a **phantom `SIGNED_OUT`** (transient refresh / network while the refresh token is still valid) both look like “the user got logged out,” but they are different:

- If storage was actually cleared, the session is gone → treat as **expired** and re-auth.
- If `onAuthStateChange` fired `SIGNED_OUT` and a durable store still has a token, **contest** it: one resolver should return `authenticated | expired | unresolved`. Do not redirect to login on network / `unresolved`.

Write-up (MIT): https://github.com/orleib-lab/agent-skills/tree/main/session-verdict  
`npx skills add orleib-lab/agent-skills --skill session-verdict`

SDK-side refresh preservation: https://github.com/supabase/supabase-js/pull/2430

**Comment by orleib-lab** [comment]
Follow-up on the earlier session-verdict note: the **phantom `SIGNED_OUT`** half (transient refresh wiping a still-valid session) was fixed in supabase-js 2.108.2 ([#2436](https://github.com/supabase/supabase-js/pull/2436)). Do not contest `SIGNED_OUT` with a second token copy on current clients.

This thread’s original issue stands: iOS WebView `localStorage` eviction. Put Auth `storage` on Capacitor Preferences (or native Auth). Empty `getSession()` after eviction is expired storage, not a phantom event.
